docker/etc/s6/gitea/setup

```diff
@@ -12,7 +12,29 @@ fi
 
 if [ ! -f /data/gitea/conf/app.ini ]; then
     mkdir -p /data/gitea/conf
-    cp /etc/templates/app.ini /data/gitea/conf/app.ini
+
+    # Set INSTALL_LOCK to true only if SECRET_KEY is not empty and
```

Comment:
This is a good question, why it's needed?

It was obviously that we alway go through the installation process to setup the `SECRET_KEY`. However, why we have to?

I didn't find any document clear the relationship between `INSTALL_LOCK` and `SECRET_KEY`, so I read the code, and I found the `SECRET_KEY` will be randomly generated ONLY during the installation:

https://github.com/go-gitea/gitea/blob/2c3a229a3c4cc3e86c5a1130bbd058ba78022a6a/routers/install.go#L315-L320

```go
	var secretKey string
	if secretKey, err = base.GetRandomString(10); err != nil {
		ctx.RenderWithErr(ctx.Tr("install.secret_key_failed", err), tplInstall, &form)
		return
	}
	cfg.Section("security").Key("SECRET_KEY").SetValue(secretKey)
```

Otherwise, it will try to find the user setting `SECRET_KEY` first, and if user is not providing the value, it will use the default string `!#@FDEWREWR&*(`, which is not safe and should be random generated.

https://github.com/go-gitea/gitea/blob/ced50e0ec13085504fa19c82f018a2eecb70ff68/modules/setting/setting.go#L813

```go
	SecretKey = sec.Key("SECRET_KEY").MustString("!#@FDEWREWR&*(")
```

ping @sapk and @lunny , could you tell me why the default value for the `SECRET_KEY` is a static value, instead of a randomly generated key?

I read the issue https://github.com/go-gitea/gitea/pull/455 , I'm still not clear. I think the `SECRET_KEY` should always be generated if the value is not provided by the user, the static default string should be avoided in this case.

Reply:
@twang2218 It should be.